packages/wallet/test/index.spec.ts

```diff
@@ -27,617 +27,659 @@ const isRunningInKarma = (): boolean => {
-      fixtureWallet.getPrivateKeyString(),
-      (await Wallet.fromV3(w, pw)).getPrivateKeyString()
-    )
-    assert.strictEqual(
-      fixtureWallet.getPrivateKeyString(),
-      (await Wallet.fromV3(w2, pw)).getPrivateKeyString()
+  } catch (err: any) {
+    t.ok(err.message.includes(errBuffLength))
+  }
+  try {
+    await fixtureWallet.toV3(pw, { uuid: {} as never as any })
+  } catch (err: any) {
+    t.ok(
+      err.message.includes(
+        'Invalid uuid, must be a string (32 hex characters) or buffer (16 bytes)'
+      )
     )
-  })
+  }
+})
+tape('should strip leading "0x" from salt, iv, uuid', async (t) => {
+  const pw = 'test'
+  const salt =
+    'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa'
+  const iv = 'bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb'
+  const uuid = 'cccccccccccccccccccccccccccccccc'
+  let w = await fixtureWallet.toV3(pw, {
+    salt: '0x' + salt,
+    iv: '0X' + iv,
+    uuid: '0x' + uuid,
+    kdf: 'pbkdf2',
+  })
+  let w2 = await fixtureWallet.toV3(pw, {
+    salt: '0x' + salt,
+    iv: '0X' + iv,
+    uuid,
+    kdf: 'pbkdf2',
+  })
+
+  t.equal(salt, w.crypto.kdfparams.salt)
+  t.equal(iv, w.crypto.cipherparams.iv)
+  t.equal(w.id, w2.id)
+  t.equal(fixtureWallet.getPrivateKeyString(), (await Wallet.fromV3(w, pw)).getPrivateKeyString())
+  t.equal(fixtureWallet.getPrivateKeyString(), (await Wallet.fromV3(w2, pw)).getPrivateKeyString())
+
+  w = await fixtureWallet.toV3(pw, {
+    salt: '0x' + salt,
+    iv: '0X' + iv,
+    uuid: '0x' + uuid,
+    kdf: 'scrypt',
+  })
+  w2 = await fixtureWallet.toV3(pw, {
+    salt: '0x' + salt,
+    iv: '0X' + iv,
+    uuid,
+    kdf: 'scrypt',
+  })
+
+  t.equal(salt, w.crypto.kdfparams.salt)
+  t.equal(iv, w.crypto.cipherparams.iv)
+  t.equal(w.id, w2.id)
+  t.equal(fixtureWallet.getPrivateKeyString(), (await Wallet.fromV3(w, pw)).getPrivateKeyString())
+  t.equal(fixtureWallet.getPrivateKeyString(), (await Wallet.fromV3(w2, pw)).getPrivateKeyString())
 })
 
 /*
-describe('.fromV1()', function () {
-  it('should work', function () {
+tape('.fromV1()',  (t) => {
+  t.test('should work', function () {
     const sample = '{"Address":"d4584b5f6229b7be90727b0fc8c6b91bb427821f","Crypto":{"CipherText":"07533e172414bfa50e99dba4a0ce603f654ebfa1ff46277c3e0c577fdc87f6bb4e4fe16c5a94ce6ce14cfa069821ef9b","IV":"16d67ba0ce5a339ff2f07951253e6ba8","KeyHeader":{"Kdf":"scrypt","KdfParams":{"DkLen":32,"N":262144,"P":1,"R":8,"SaltLen":32},"Version":"1"},"MAC":"8ccded24da2e99a11d48cda146f9cc8213eb423e2ea0d8427f41c3be414424dd","Salt":"06870e5e6a24e183a5c807bd1c43afd86d573f7db303ff4853d135cd0fd3fe91"},"Id":"0498f19a-59db-4d54-ac95-33901b4f1870","Version":"1"}'
     const wallet = Wallet.fromV1(sample, 'foo')
-    assert.strictEqual(wallet.getAddressString(), '0xd4584b5f6229b7be90727b0fc8c6b91bb427821f')
+    t.deepEqual(wallet.getAddressString(), '0xd4584b5f6229b7be90727b0fc8c6b91bb427821f')
   })
 })
 */
 
```

Comment:
Do we still need to keep this commented out test?

Reply:
Not sure.  Let me look at the comments from @paulmillr when the big upgrade was done a few months back.  It fails if I uncomment it so not sure if it's necessary or not.

Reply:
Not sure.  Let me look at the comments from @paulmillr when the big upgrade was done a few months back.  It fails if I uncomment it so not sure if it's necessary or not.